ethsigner/signer/hashicorp/src/main/java/tech/pegasys/ethsigner/signer/hashicorp/HashicorpConfig.java

```diff
@@ -56,13 +63,23 @@ public String getSigningKeyPath() {
     return signingKeyPath;
   }
 
+  public boolean isTlsEnabled() {
+    return tlsEnabled;
+  }
+
+  public Optional<PkcsTrustStoreConfig> getPkcsTrustStoreConfig() {
+    return Optional.ofNullable(pkcsTrustStoreConfig);
```

Comment:
IntelliJ complains (show yellow marker) if I store optional as class variable (I think because Optional is not serializable). However, returning Optional from getter is perfectly acceptable.

Reply:
IntelliJ has a really bad check which says you shouldn't use `Optional` as anything but a return type.  It's based on an email from one of the designers of the Stream API very early on when it was expected that `Optional` would be a temporary thing which could be optimised by future JVM improvements.

The world (apart from IntelliJ) has long since moved on from this bad advice and you should disable that check in IntelliJ. :)

Reply:
:) good to know.

Reply:
this bit is gone, but the new stuff is using Optional as member variable.